### PR TITLE
Fix external media url uploads on atomic

### LINKED ISSUE #17412: Media: URL upload broken on Atomic sites
#### Steps to reproduce the issue

1. Go to My Site -> Site -> Media
2. Click on the arrow next to the "Add new" button and choose "Add via URL"
3. Add an image URL and you will get "1 file could not be uploaded because an error occurred while uploading." error.

#### What I expected

For the image to be uploaded like on sites which do not have custom plugins installed. 

#### What happened instead

It seems this issue may be happening because of this PR - #16986.

A stack trace:

```
[08-Oct-2020 05:52:34 UTC] PHP Fatal error:  Uncaught Error: Call to undefined function get_upload_space_available() in /wordpress/plugins/jetpack/9.0.1/json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php:185
Stack trace:
#0 /wordpress/core/5.5.1/wp-includes/class-wp-hook.php(287): WPCOM_JSON_API_Upload_Media_v1_1_Endpoint->check_upload_size(Array)
#1 /wordpress/core/5.5.1/wp-includes/plugin.php(206): WP_Hook->apply_filters(Array, Array)
#2 /wordpress/core/5.5.1/wp-admin/includes/file.php(754): apply_filters('wp_handle_sidel...', Array)
#3 /wordpress/core/5.5.1/wp-admin/includes/file.php(1018): _wp_handle_upload(Array, Array, '2020-10-07 16:4...', 'wp_handle_sidel...')
#4 /wordpress/core/5.5.1/wp-admin/includes/media.php(449): wp_handle_sideload(Array, Array, '2020-10-07 16:4...')
#5 /wordpress/plugins/jetpack/9.0.1/class.json-api-endpoints.php(1943): media_handle_sideload(Array, 45533)
#6 /wordpress/plugins/jetpack/9.0.1/class.json-api-endpoints.php(1836): WPCOM_JSON_API_Endpoint->handle_media_sideload('https://cldup.c...', 45533)
#7 /wordpress/plugins/jetpack/9.0.1/json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php(114): WPCOM_JSON_API_Endpoint->handle_media_creation_v1_1(Array, Array, Array)
#8 /wordpress/plugins/jetpack/9.0.1/class.json-api.php(348): WPCOM_JSON_API_Upload_Media_v1_1_Endpoint->callback('/sites/16514019...', 1)
#9 /wordpress/plugins/jetpack/9.0.1/class.json-api.php(332): WPCOM_JSON_API->process_request(Object(WPCOM_JSON_API_Upload_Media_v1_1_Endpoint), Array)
#10 /wordpress/plugins/jetpack/9.0.1/vendor/automattic/jetpack-connection/legacy/class-jetpack-xmlrpc-server.php(920): WPCOM_JSON_API->serve(false)
#11 /wordpress/core/5.5.1/wp-includes/IXR/class-IXR-server.php(121): Jetpack_XMLRPC_Server->json_api(Array)
#12 /wordpress/core/5.5.1/wp-includes/IXR/class-IXR-server.php(206): IXR_Server->call('jetpack.jsonAPI', Array)
#13 /wordpress/core/5.5.1/wp-includes/IXR/class-IXR-server.php(109): IXR_Server->multiCall(Array)
#14 /wordpress/core/5.5.1/wp-includes/IXR/class-IXR-server.php(59): IXR_Server->call('system.multical...', Array)
#15 /wordpress/core/5.5.1/wp-includes/IXR/class-IXR-server.php(27): IXR_Server->serve('<?xml version="...')
#16 /wordpress/core/5.5.1/wp-includes/IXR/class-IXR-server.php(35): IXR_Server->__construct(Array, false, false)
#17 /wordpress/core/5.5.1/wp-includes/class-wp-xmlrpc-server.php(191): IXR_Server->IXR_Server(Array)
#18 /wordpress/core/5.5.1/xmlrpc.php(87): wp_xmlrpc_server->serve_request()
#19 {main}
  thrown in /wordpress/plugins/jetpack/9.0.1/json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php on line 185
```

p1602136996077900-slack-jetpack-developers

### PR DESCRIPTION
**Pull request by cpapazoglou**
Fixes #17412

#### Changes proposed in this Pull Request:

* Since `get_upload_space_available` is not available in non multisites run space check only in multisites (eg simple sites)

#### Testing instructions:

We should test two cases, Atomic & Simple Sites

### Atomic:
**Before Applying this patch**
* Launch an ephemeral site
* Connect jetpack to wpcom
* Go to `https://wordpress.com/media/`
* Try adding an image via URL
* You should get an error

**After Applying this patch to your ephemeral site via ssh**
* Go to `https://wordpress.com/media/`
* Try adding an image via URL
* Media should be uploaded as expected

### Simple Sites:
**On wpcom simple sites**
* Apply D50839-code
* Go to `https://wordpress.com/media/` of a simple site
* Try adding an image via URL
* Media should be uploaded as expected

#### Proposed changelog entry for your changes:

* Uploading media from URL fix

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php [hunk 1]
```diff
@@ -108,8 +108,10 @@ function callback( $path = '', $blog_id = 0 ) {
 
 		// Normal WPCOM upload processing
 		if ( count( $other_media_files ) > 0 || count( $media_urls ) > 0 ) {
-			add_filter( 'wp_handle_upload_prefilter', array( $this, 'check_upload_size' ), 9 ); // used for direct media uploads.
-			add_filter( 'wp_handle_sideload_prefilter', array( $this, 'check_upload_size' ), 9 ); // used for uploading media via url.
+			if ( is_multisite() ) { // Do not check for available space in non multisites.
```
Comment (aidvu) [inline json-endpoints/class.wpcom-json-api-upload-media-v1-1-endpoint.php:111]: Makes sense. 👍🏻

**Review by aidvu — APPROVED** [review]
lgtm

**Review by jeherve — APPROVED** [review]
This tests well for me. 🚢

**Comment by cpapazoglou** [comment]
Deployed r214892-wpcom

**Comment by jeherve** [comment]
Cherry-picked to `branch-9.0` in 2e956f8c2c902dd7b0bf1a12b5506bdaf3dce1a8

**Pull request merged**